recipes/fontconfig/all/conanfile.py

```diff
@@ -16,39 +16,52 @@ class FontconfigConan(ConanFile):
     options = {"shared": [True, False], "fPIC": [True, False]}
     default_options = {"shared": False, "fPIC": True}
     generators = "pkg_config"
-    _source_subfolder = "source_subfolder"
+
     _autotools = None
 
-    def requirements(self):
-        self.requires("freetype/2.10.2")
-        self.requires("expat/2.2.9")
-        if self.settings.os == "Linux":
-            self.requires("libuuid/1.0.3")
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
-    def configure(self):
+    def config_options(self):
         if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds are not supported.")
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Visual Studio builds are not supported.")
+        if self.options.shared:
+            del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+    def requirements(self):
+        self.requires("freetype/2.10.4")
+        self.requires("expat/2.2.10")
+        if self.settings.os == "Linux":
+            self.requires("libuuid/1.0.3")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
+        else:
+            self.build_requires("gperf/3.1")
```

Comment:
Is `gperf` required on mingw@Windows?

Reply:
msys2 recipe already prodives gperf.

Reply:
But when using e.g. wsl2, and setting `CONAN_BASH_PATH`, then gperf is not available.

Reply:
It is available. If `tools.get_env("CONAN_BASH_PATH")` is evaluated to True, `tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH")` is False, then `gperf` is added to build requirements.

Reply:
This depends on the `additional_packages` option of `msys2`.

https://github.com/conan-io/conan-center-index/blob/8f8041272d0284459d86993b6aa73e0bd4a6d8a8/recipes/msys2/all/conanfile.py#L20
I think it's safer to just add gperf unconditionally.
We would also add bison/flex to the build requirements.

Reply:
Missed that, ok I'll add gperf.
Why flex/bison?

Reply:
I mentioned flex/bison as an example.
I'm not on windows atm, but I think msys2 also includes those packages.